### PR TITLE
ci(security): Upload Semgrep & Trivy scan results to GitHub Code Scanning in PRs

### PR DESCRIPTION
**Pull request by tomkerkhove**
Upload Semgrep & Trivy results to GitHub Code Scanning in PRs so that they will be depicted in the PR checks:

image

### Checklist

- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

### CODE REVIEW

**Comment by tomkerkhove** [comment]
I'll fix DCO when it passes

**Review by JorTurFer — COMMENTED** [review]
Have you changed anything in the workflows? I guess that there are missing changes

**Comment by tomkerkhove** [comment]
@JorTurFer Can you elaborate on what you think is missing?

**Comment by tomkerkhove** [comment]
@zroubalik any reason why this was closed?

**Comment by tomkerkhove** [comment]
No worries at all! I still want to do it, but got dropped off my  todo list, soryr.

File: CHANGELOG.md [hunk 1]
```diff
@@ -56,6 +56,8 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 - **External Scaler**: Add tls options in TriggerAuth metadata. ([#3565](https://github.com/kedacore/keda/issues/3565))
 - **GCP PubSub Scaler**: Make it more flexible for metrics ([#4243](https://github.com/kedacore/keda/issues/4243))
 - **Kafka Scaler:** Add support for OAuth extensions ([#4544](https://github.com/kedacore/keda/issues/4544))
+- **Security**: Automatically publish Semgrep scans to GitHub Code Scanning for PRs ([#4560](https://github.com/kedacore/keda/pull/4560))
```
Comment (zroubalik) [inline CHANGELOG.md:59]: This should be in the Other section, as it is not user facing change

**Review by zroubalik — COMMENTED** [review]
Do we have any update here @tomkerkhove ?

**Comment by tomkerkhove** [comment]
I failed the force push which deleted the changes so abandoning everything

**Pull request closed without merging**